### PR TITLE
common: modify 'main()s' to use new argv_to_vec() signature 

### PR DESCRIPTION
**Pull request by ronen-fr**
A followup to PR #42820 that modified argv_to_vec() signature
(for style and performance).

Also - a minor change to argv_to_vec() signature, as    
some of Ceph main()'s use the standard 'char** av', while others use
'const char**'. The modified signature of argv_to_vec() handles both,
allowing us to remove the existing const-cast used by some executables.

Signed-off-by: Ronen Friedman <rfriedma@redhat.com>

### CODE REVIEW

**Review by tchaikov — APPROVED** [review]

File: src/test/common/get_command_descriptions.cc [hunk 1]
```diff
@@ -97,8 +97,7 @@ static void pull585()
 }
 
 int main(int argc, char **argv) {
-  vector<const char*> args;
-  argv_to_vec(argc, (const char **)argv, args);
+  auto args = argv_to_vec(argc, argv);
```
Comment (tchaikov) [inline src/test/common/get_command_descriptions.cc:100]:
```
../src/test/common/get_command_descriptions.cc:100:15: error: no matching function for call to 'argv_to_vec'
  auto args = argv_to_vec(argc, argv);
              ^~~~~~~~~~~
../src/common/ceph_argparse.h:49:33: note: candidate function not viable: no known conversion from 'char **' to 'const char **' for 2nd argument
extern std::vector<const char*> argv_to_vec(int argc, const char** argv);
                                ^
```

do we need a variant which accepts `char **` well?

Comment (ronen-fr) [reply src/test/common/get_command_descriptions.cc:100]: I made a mistake when reordering the commits (that's why I didn't yet remove the draft status).
Almost finished cleaning up now.

**Review by tchaikov — APPROVED** [review]

File: src/ceph_fuse.cc [hunk 2]
```diff
@@ -92,8 +92,7 @@ void usage()
 int main(int argc, const char **argv, const char *envp[]) {
   int filer_flags = 0;
   //cerr << "ceph-fuse starting " << myrank << "/" << world << std::endl;
-  std::vector<const char*> args;
-  argv_to_vec(argc, argv, args);
+  auto args = argv_to_vec(argc, argv);
```
Comment (rzarzynski) [inline src/ceph_fuse.cc:95]: It looks far more natural now.

**Review by rzarzynski — APPROVED** [review]
LGMT!

File: src/common/ceph_argparse.cc [hunk 3]
```diff
@@ -126,13 +126,7 @@ void env_to_vec(std::vector<const char*>& args, const char *name)
...
 }
 
-void argv_to_vec(int argc, const char **argv,
-                 std::vector<const char*>& args)
-{
-  args.insert(args.end(), argv + 1, argv + argc);
-}
-
-std::vector<const char*> argv_to_vec(int argc, const char** argv)
+std::vector<const char*> argv_to_vec(int argc, const char* const * argv)
```
[the remark this answers is not in the document]
Comment (ronen-fr) [reply src/common/ceph_argparse.cc:129]: ... and that is just the beginning!

**Comment by ideepika** [comment]
@ronen-fr can you take a final look and merge if it looks good to you:
https://pulpito.ceph.com/?branch=wip-deepika2-testing-2021-08-27-0331
https://tracker.ceph.com/issues/52398
https://tracker.ceph.com/issues/49287
https://tracker.ceph.com/issues/52315
https://tracker.ceph.com/issues/52447

**Comment by ronen-fr** [comment]
> @ronen-fr can you take a final look and merge if it looks good to you:
> https://pulpito.ceph.com/?branch=wip-deepika2-testing-2021-08-27-0331
> https://tracker.ceph.com/issues/52398
> https://tracker.ceph.com/issues/49287
> https://tracker.ceph.com/issues/52315
> https://tracker.ceph.com/issues/52447

Thanks a lot for the effort!
checking

**Pull request merged**